code/game/objects/machinery/squad_supply/supply_console.dm

```diff
@@ -6,55 +6,64 @@
 	icon_state = "supplydrop"
 	interaction_flags = INTERACT_MACHINE_TGUI
 	///Time between two supply drops
-	var/launch_cooldown = 2 MINUTES
+	var/launch_cooldown = 1 MINUTES
 	COOLDOWN_DECLARE(next_fire)
-	///If true, the last squad will be memorised
-	var/squad_lock = null
-	///The squad wich we will look for beacons
-	var/datum/squad/current_squad = null
 	///The beacon we will send the supplies
-	var/obj/item/squad_beacon/current_beacon = null
+	var/datum/supply_beacon/current_beacon = null
 	///The content sent
 	var/list/supplies = list()
 	///X offset of the drop, relative to the supply beacon loc
 	var/x_offset = 0
 	///Y offset of the drop, relative to the supply beacon loc
 	var/y_offset = 0
+	/// The pad it is linked to
+	var/obj/structure/supply_drop/drop_pad
 
 /obj/machinery/computer/supplydrop_console/Initialize()
 	. = ..()
-	if(!isnull(squad_lock))
-		current_squad = SSjob.squads[squad_lock]
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/computer/supplydrop_console/LateInitialize()
+	. = ..()
+	drop_pad = GLOB.supply_pad
 
 /obj/machinery/computer/supplydrop_console/Destroy()
-	current_squad = null
 	current_beacon = null
 	return ..()
 
 /obj/machinery/computer/supplydrop_console/ui_interact(mob/user, datum/tgui/ui)
 	. = ..()
 	ui = SStgui.try_update_ui(user, src, ui)
+	if(!current_beacon?.registered)
+		set_current_beacon()
+		if(!current_beacon)
+			to_chat(user, "<span class='warning'>No active beacon was detected!</span>")
+			ui?.close()
+			return
 	if (!ui)
 		ui = new(user, src, "SupplyDropConsole", name)
 		ui.open()
 
-
-/obj/machinery/computer/supplydrop_console/ui_static_data(mob/user)
-	. = ..()
-	.["squads"] = list()
-	for(var/squad in SSjob.active_squads)
-		.["squads"] += list(squad)
+/obj/machinery/computer/supplydrop_console/proc/set_current_beacon(datum/source, datum/supply_beacon/beacon)
+	SIGNAL_HANDLER
+	if(current_beacon)
+		UnregisterSignal(current_beacon, COMSIG_BEACON_UNREGISTERED)
+	if(!beacon)
+		if(!length(GLOB.supply_beacon_list))
+			current_beacon = null
+			return
+		beacon = GLOB.supply_beacon_list[1]
+	current_beacon = beacon
+	RegisterSignal(current_beacon, COMSIG_BEACON_UNREGISTERED, .proc/set_current_beacon)
 
 /obj/machinery/computer/supplydrop_console/ui_data(mob/user)
 	. = ..()
-	.["squad_lock"] = squad_lock
 	.["launch_cooldown"] = launch_cooldown
-	.["current_squad"] = current_squad?.name
-	.["current_beacon"] = list(
-		"name" = current_squad?.sbeacon?.name,
-		"x_coords" = current_squad?.sbeacon?.loc.x,
-		"y_coords" = current_squad?.sbeacon?.loc.y
-	)
+	.["current_beacon"] = current_beacon
+	.["beacon_x"] = current_beacon?.location.x
+	.["beacon_y"] = current_beacon?.location.y
+	.["beacon_name"] = current_beacon?.name 
+	.["beacon_registered"] = current_beacon?.registered
```

Comment:
these vars should be static data, only updated when beacon changes.

Reply:
See splinters review